### PR TITLE
[Fixes AB#808] JWT renewal

### PR DESCRIPTION
**Pull request by mightymatth**

### CODE REVIEW

File: src/app/shared/services/error.service.ts [hunk 1]
```diff
@@ -83,11 +83,21 @@ export class ErrorService {
...
+                            ));
+                            break;
+                        }
+                    // tslint:disable-next-line:no-switch-case-fall-through
                     case AuthError.MISSING_JWT:
                     case AuthError.CANNOT_REGISTER_JWT:
                     case UserError.USER_JWT_MISSING:
-                        action$ = this.takeAction(() => of(this.userService.logout())
-                            .pipe(this.router.navigate['/']));
+                    case undefined: // TODO: backend returns 400 with no err_code. fix this. this is workaround.
```
Comment (mightymatth) [inline src/app/shared/services/error.service.ts:97]: @githinho,
When I try to get a new access token with refresh token route (POST `/api/user/token/refresh`), I get HTTP error 400, but the `err_code` field is empty. Can I get some code here to be consistent with the rest of the responses?

Comment (githinho) [reply src/app/shared/services/error.service.ts]: The code is 0208: https://github.com/AMPnet/docs/blob/master/ErrorCodes.md#authentication---prefix-02

**Review by githinho — COMMENTED** [review]
Very nice cleanup with error handling. I will test this tomorrow.

**Review by githinho — COMMENTED** [review]
I have found a problem with generating reports when the refresh token has expired, the response is 400, but the application doesn't log me out. 
`{"description":"Invalid JWT","err_code":"0204","message":"Invalid JWT"}`
The rest is working smoothly.

**Pull request merged**